### PR TITLE
Add instructions on how to use the official Twig extensions repo

### PR DESCRIPTION
**Pull request by mpdude**
There are several references to the official Twig extensions repository, but we don't mention how easy it actually is to use that.

When #7613 is merged we also should update the notes added there to include a reference to this new section.

### CODE REVIEW

File: templating/twig_extension.rst [hunk 1]
```diff
@@ -15,8 +15,12 @@ your code faster.
 
 .. tip::
 
-    Before writing your own extensions, have a look at the
-    `Twig official extension repository`_.
+    When writing your own extensions, you might want to learn
+    from having a look at the `Twig Bridge`_  which contains most of
+    the extensions provided by the Symfony Framework.
+
+    We also have :doc:`a short article </reference/using_twig_extension_repository>`
```
Comment (xabbuh) [inline templating/twig_extension.rst]: `/templating/using_twig_extension_repository`

Comment (xabbuh) [inline templating/twig_extension.rst]: maybe we should also rename the new file to something shorter like `/templating/twig_extension_repository`

Comment (mpdude) [inline templating/twig_extension.rst]: Fixed

File: templating/twig_extension_repository.rst [hunk 2]
```diff
@@ -40,7 +40,7 @@ Then, define the extension class as a service and tag it with the
 
         # app/config/services.yml
         services:
-            twig_extension.intl:
+            twig_extensions.intl:
```
Comment (HeahDude) [inline templating/twig_extension_repository.rst]: The namespace should not have an "s" IMHO, what do others think?

Comment (mpdude) [inline templating/twig_extension_repository.rst]: I chose that because the repo is called "Twig Extension*s*". Thus I'd call a corresponding bundle "TwigExtensionsBundle", which would give this name. 

Or should we better recommend prefixing it with `app`?

Comment (xabbuh) [inline templating/twig_extension_repository.rst]: I guess `app.` is better since there may be other bundles who try the same that we do here.

Comment (mpdude) [inline templating/twig_extension_repository.rst]: Now that you mention it... For example, I've come across [this one](https://github.com/rdohms/dms-twig-extension-bundle/blob/master/README.md).

Would you guys recommend using a 3rd party bundle instead of providing instructions?

Note I am not affiliated with that bundle in any way and cannot endorse it.

Comment (xabbuh) [inline templating/twig_extension_repository.rst]: I would always strive to keep the number of external dependencies as low as possible. Given that registering services is not hard and does not require much work I would only require that bundle if you need one of the extensions shipped with it.

File: templating/twig_extension.rst [hunk 3]
```diff
@@ -15,8 +15,14 @@ your code faster.
 
 .. tip::
 
-    Before writing your own extensions, have a look at the
-    `Twig official extension repository`_.
+    When writing your own extensions, you might want to learn
+    from having a look at the `Twig Bridge`_  which contains most of
+    the extensions provided by the Symfony Framework.
+
+.. tip::
```
Comment (xabbuh) [inline templating/twig_extension.rst]: we could use `.. seealso::` instead here

File: templating/twig_extension.rst [hunk 4]
```diff
@@ -15,8 +15,14 @@ your code faster.
 
 .. tip::
 
-    Before writing your own extensions, have a look at the
-    `Twig official extension repository`_.
+    When writing your own extensions, you might want to learn
+    from having a look at the `Twig Bridge`_  which contains most of
+    the extensions provided by the Symfony Framework.
```
Comment (xabbuh) [inline templating/twig_extension.rst]: framework

File: templating/twig_extension.rst [hunk 5]
```diff
@@ -15,8 +15,14 @@ your code faster.
...
+
+.. tip::
+
+    We also have :doc:`a short article </templating/twig_extension_repository>`
```
Comment (xabbuh) [inline templating/twig_extension.rst]: We always avoid the first person perspective in the docs. This could be rewritten to something like "Read ... on how to use [...]"

File: templating/twig_extension_repository.rst [hunk 6]
```diff
@@ -0,0 +1,92 @@
+.. index::
+   single: Using the Twig Extensions Repository
+
+How to Use the Twig Extensions Repository
+=========================================
+
+The `Twig official extension repository`_ contains (as of writing) some
```
Comment (xabbuh) [inline templating/twig_extension_repository.rst]: The official Twig extensions repository [...]

Comment (xabbuh) [inline templating/twig_extension_repository.rst]: I think we can drop the "as of writing" part

File: templating/twig_extension_repository.rst [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+helpful Twig extensions that are not part of the Twig core. They add
+useful functions for internationalization, working with arrays and
+dates. To learn more about these extensions, have a look at their
+`documentation`_.
+
+This repository is meant as an extension to Twig in general. So, it
+is does *not* provide a direct means to register itself with the
```
Comment (xabbuh) [inline templating/twig_extension_repository.rst]: So, it ~~is~~ does not [...]

File: templating/twig_extension_repository.rst [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+Symfony Framework (it is not a Bundle).
```
Comment (xabbuh) [inline templating/twig_extension_repository.rst]: framework

Comment (xabbuh) [inline templating/twig_extension_repository.rst]: bundle

File: templating/twig_extension_repository.rst [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+
+It is, however, very easy to get the extensions set-up in Symfony.
```
Comment (xabbuh) [inline templating/twig_extension_repository.rst]: set up

File: templating/twig_extension_repository.rst [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+that repository so you can use it in your Twig templates.
+
+.. tip::
+
+    Setting up one of the other extensions works just the same way,
+    except you need to choose another service id and have to use
+    the right class name.
+
+First, add the Twig Extensions repository as a dependency in your
+project. Assuming you are using Composer, run
```
Comment (xabbuh) [inline templating/twig_extension_repository.rst]: IMO we should ignore that in theory there are other ways to install the extensions and reword this to "First, install the Twig extensions using Composer`.

File: templating/twig_extension_repository.rst [hunk 11]
```diff
@@ -0,0 +1,92 @@
...
+                    class="Twig_Extensions_Extension_Intl">
+                    <tag name="twig.extension" />
+                </service>
+            </services>
+        </container>
+
+    .. code-block:: php
+
+        // app/config/services.php
+        use \Twig_Extensions_Extension_Intl;
```
Comment (xabbuh) [inline templating/twig_extension_repository.rst]: In Symfony, we usually don't have `use` statements for classes from the global namespace.

File: templating/twig_extension_repository.rst [hunk 12]
```diff
@@ -0,0 +1,92 @@
...
+
+        $container
+            ->register('app.twig_extensions.intl', Twig_Extensions_Extension_Intl::class)
+            ->addTag('twig.extension');
+
+And that's it! For example, you should now be able to use the
+``localizeddate`` filter to format a date according to the request's
+current locale:
```
Comment (xabbuh) [inline templating/twig_extension_repository.rst]: [...] to the current request's locale

File: templating/twig_extension_repository.rst [hunk 13]
```diff
@@ -0,0 +1,92 @@
...
+    {{ post.published_at|localizeddate('medium', 'none', locale) }}
+
+Learning further
+----------------
+
+In the :doc:`reference section </reference/twig_reference>`, you can
+find all the extra Twig functions, filters, tags and tests that are
+already added by the Symfony Framework.
+
+We also have documentation on :doc:`how to write your own Twig extension </templating/twig_extension>`.
```
Comment (xabbuh) [inline templating/twig_extension_repository.rst]: first person :)

File: templating/twig_extension_repository.rst [hunk 14]
```diff
@@ -0,0 +1,92 @@
...
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # app/config/services.yml
+        services:
+            app.twig_extensions.intl:
+                class: Twig_Extensions_Extension_Intl
+                tags:
+                    - { name: twig.extension }
```
Comment (xabbuh) [inline templating/twig_extension_repository.rst]: maybe also mark the service as private

**Comment by mpdude** [comment]
@xabbuh Thanks for the thorough review, all fixed!

**Review by roman-1983 — APPROVED** [review]

**Comment by mpdude** [comment]
@HeahDude / @xabbuh any other wishes/remarks?

File: templating/twig_extension_repository.rst [hunk 15]
```diff
@@ -0,0 +1,93 @@
+.. index::
+   single: Using the Twig Extensions Repository
+
+How to Use the Twig Extensions Repository
+=========================================
+
+The `Twig official extensions repository`_ contains some
```
Comment (xabbuh) [inline templating/twig_extension_repository.rst]: Twig official extensions repository (or change the target at the end of the document)

Comment (mpdude) [inline templating/twig_extension_repository.rst]: Fixed link below

**Comment by xabbuh** [comment]
:+1: the build is passing now :)

Status: Reviewed

File: templating/twig_extension_repository.rst [hunk 16]
```diff
@@ -0,0 +1,93 @@
+.. index::
+   single: Using the Twig Extensions Repository
+
+How to Use the Twig Extensions Repository
```
Comment (javiereguiluz) [inline templating/twig_extension_repository.rst]: Can we change the title of this article?  The `Repository` word is confusing here. What we want is to enable/use the official Twig extensions in Symfony apps.

File: templating/twig_extension_repository.rst [hunk 17]
```diff
@@ -0,0 +1,93 @@
+.. index::
+   single: Using the Twig Extensions Repository
```
Comment (javiereguiluz) [inline templating/twig_extension_repository.rst]: I wonder if we really need this article. We don't explain those extensions (because they are documented in their own repo) and they are not special regarding the Symfony+Twig config: they are enabled just like any other Twig extensions.

Comment (mpdude) [inline templating/twig_extension_repository.rst]: You're right in that there is nothing special with these extensions.

However, all Twig Extensions I normally use are either part of a bundle or there is a bundle to set them up in Symfony.

I felt there was kind of a gap for the "official" Twig Extensions. They're mentioned in the Symfony docs, but every time I need them it takes me moment to remind myself that I have to set them up that way - which is why I wrote this article to help the future me :-).

File: templating/twig_extension_repository.rst (hunk 16 above)
Comment (mpdude) [inline templating/twig_extension_repository.rst]: Unfortunately, "The official Twig extensions" isn't very helpful either, rather misleading...

Comment (mpdude) [inline templating/twig_extension_repository.rst]: "Using Twig extensions not provided as a bundle"...? Meh.

Comment (HeahDude) [inline templating/twig_extension_repository.rst]: What about repeating what's below and just add "official":
```
How to Use the Twig Official Extensions
```

**Comment by mpdude** [comment]
So guys, please decide if you'd merge this or not... If you want to, then I will have a look at the conflicts.

**Comment by robfrawley** [comment]
:+1: for adding this documentation of the "Official Twig Extension" usage.

**Comment by weaverryan** [comment]
Sorry for the delay on this @mpdude - it's a good PR as usual from you. Because things have changed (I'm talking about the introduction to Flex), my vote would be to *not* merge this. Well, actually, I *do* think we should mention (somewhere) that this library *exists* with details on how to install it. But in Flex, it's as simple as running `composer require` and then un-commenting the extension you want.

So, should we merge this PR to older branches, or just update it with the simpler instructions and merge to master?

**Comment by mpdude** [comment]
Let's look ahead and not merge into old branches.

However, I haven't yet worked with Flex myself, so I am unsure if I can be of any help here?

**Comment by weaverryan** [comment]
@mpdude I tend to agree :). When you install the twig extensions repo in flex, you're given this file: https://github.com/symfony/recipes/blob/master/twig/extensions/1.0/config/packages/twig_extensions.yaml. So basically, the user only needs to go to this file and uncomment the extensions they want :). I think updating this PR (and just removing the stuff that's not needed anymore) is the easiest way forward.

**Comment by javiereguiluz** [comment]
To move things forward, I propose to close this in favor of #9035. Thanks!
